### PR TITLE
LuaRocks API

### PR DESCRIPTION
**Pull request by jiteshpabla**

### CODE REVIEW

**Comment by jiteshpabla** [comment]
@hishamhm I do not know what is causing the conflict now, all the command line instructions said that everything up-to-date.

**Comment by hishamhm** [comment]
@jiteshpabla there are *hundreds* of commits in this PR now, it's really broken — what did you do?

Since I still have locally the cleaned up version I prepared for you in our call on Friday, my main question now is: did you make any further changes to the API module since that call? (it's hard to follow with the PR in this state.)

**Comment by jiteshpabla** [comment]
@hishamhm Since the luarocks:luarocks-api branch is totally out-of-date, adding in the commits from the master branch caused this commit explosion.
And yes, I have made changes after rebasing, to update the API based on the changes made to the CLI code.

**Comment by hishamhm** [comment]
@jiteshpabla Ah, ok. I fixed this by clicking "Edit" at the top of this page and changing the base branch to "master". The other commits have gone away. :)

**Comment by hishamhm** [comment]
Merged manually into the `luarocks-api` branch of the main repo! We can take on the cleanup and code deduplication from here! Thank you @jiteshpabla !
